submissions/7SergSmith7/Tiny JS World/index.js

```diff
@@ -0,0 +1,125 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: _put repo URL here_
+   Web app: _put project's github pages URL here_
+   */
+
+// ======== OBJECTS DEFINITIONS ========
+// Define your objects here
+class Habitant {
+  constructor(species, name, gender, phrase, friends = [], legs, hands) {
+    this.species = species;
+    this.legs = legs;
+    this.hands = hands;
+    this.name = name;
+    this.gender = gender;
+    this.phrase = phrase;
+    this.friends = friends;
+    //this.addFriends(friends);
+  }
+
+  addFriends(newFriend) {
+    this.friends.push(newFriend);
+  }
+
+  listFriends() {
+    return this.friends.length ? this.friends.join(", ") : "I am loner";
+  }
+
+  sayPhrase() {
+    return this.phrase;
+  }
+
+  toString() {
+    return `Hi! I am <strong>${this.species}</strong>, my name is <strong>${
+      this.name
+    }</strong>, my gender is <strong>${this.gender}</strong>, I have <strong>${
+      this.legs
+    }</strong> legs and <strong>${this.hands}</strong> hands. 
+     My friends: <strong>${this.listFriends()}</strong> and my phrase: <strong>${this.sayPhrase()}</strong>`;
+  }
```

Comment:
1. Too much code repetition, and there is a pattern for each property. Use list of props and `.map` to wrap each prop value in `<strong>`.

2. If Habitant has no hands then `toString` method shouldn't handle it at all. You have overloaded `toString` in the child classes for this.

Reply:
done

Reply:
Well, not exactly. You need to resolve (1) while keep (2).
Item 1 was a bit overdone. You had explicit props used, now you have no explicit list and base class manages all properties. What if a child class introduces a property it doesn't want to be a part of an output?

Hint: `['propA', 'propB'].map(propName => this[propName])`

Reply:
maybe done)